ouroboros-network/test/CDDL.hs

```diff
@@ -135,9 +150,49 @@ decodeMsg (tag, input) = case tag of
             , runPingPong (ServerAgency PingPong.TokBusy)
             ]
 
-        runBlockFetch = run (codecBlockFetch Serialise.encode Serialise.encode Serialise.decode Serialise.decode :: MonoCodec BF)
+        runBlockFetch = run (codecBlockFetch Serialise.encode Serialise.encode Serialise.decode
+                                             Serialise.decode :: MonoCodec BF)
         blockFetchParsers = [
               runBlockFetch (ClientAgency BlockFetch.TokIdle)
             , runBlockFetch (ServerAgency BlockFetch.TokBusy)
             , runBlockFetch (ServerAgency BlockFetch.TokStreaming)
             ]
+
+data Protocol ps where
+    CS :: Protocol (ChainSync BlockHeader (Point BlockHeader))
+    ReqResp :: Protocol RR
+
+data Msg where
+    Msg :: Protocol ps -> PeerHasAgency pr (st :: ps) -> Message ps (st :: ps) (st' :: ps) -> Msg
```

Comment:
Why do we need to frame all protocols together? Isn't it enough if we have each protocol tested separately? You could  write a polymorphic test which is instantiated with different protocol in a similar way to `Network.TypedProtocols.Codec.prop_codec` function.

Reply:
Yes you are right.
That is much simpler and much nicer to have separate tests for each protocol.

I can't remember exatly why I did it like that.
Maybe it would be nice to have a type that contains all protocols for the other direction
because thats how messages.cddl works